### ISSUE TITLE
WPSException Unsupported Encoding UTF-8 for Unspecfied Encoding

### ISSUE THREAD

**Issue by mlacayoemery** [geopython/OWSLib#287]
When trying to make a WPS execute call, it seems that for inputs that do not have a specified coding (UTF-8, etc) in the DescribeProcess results in a owslib.wps.WPSException Unsupported encoding UTF-8 error. This seems to be the case with PyWPS4 and Geoserver 2.8.0, so I think it is OWSLib 0.9.2. In the WPS 1.0.0 specification (Table 23), it seems that the encoding and schema fields are optional (when the input is using the same encoding as the responses), and indeed if I manually do the DescribeProcess I can see in the XML result that the encoding and schema fields are not included. Is OWSLib perhaps over enforcing the encoding requirements? If you can provide me with a WPS service URL I would gladly put together an example.

**Comment by mlacayoemery** [comment]
This still seems to be the case with OWSLib 0.11.2.

**Comment by mlacayoemery** [comment]
This still seems to be the case with OWSLib 0.16.0.

**Comment by tomkralidis** [comment]
Can you provide an example that we can use to reproduce?

**Comment by mlacayoemery** [comment]
Hi Tom,

I'm using OWSLib 0.16.0 with [GeoServer 2.12.0](https://sourceforge.net/projects/geoserver/files/GeoServer/2.12.0/geoserver-2.12.0-war.zip/download) and the [WPS plugin](https://sourceforge.net/projects/geoserver/files/GeoServer/2.12.0/extensions/geoserver-2.12.0-wps-plugin.zip/download) and trying to run the following code:

```
import owslib.wps

wps_url = "http://127.0.0.1:8080/geoserver/ows"
wps = owslib.wps.WebProcessingService(wps_url, verbose=False, skip_caps=True)

processid = "JTS:area"

polygon = [(-102.8184, 39.5273), (-102.8184, 37.418),
           (-101.2363, 37.418), (-101.2363, 39.5273),
           (-102.8184, 39.5273)]
featureCollection = owslib.wps.GMLMultiPolygonFeatureCollection( [polygon] )

inputs = [ ("geom", featureCollection)]
output = "OUTPUT"

execution = wps.execute(processid, inputs, output = "OUTPUT")

owslib.wps.monitorExecution(execution)

```

what I get is the following:

`owslib.wps.WPSException : {'code': 'NoApplicableCode', 'locator': None, 'text': 'Unsupported encoding UTF-8'}`

I'm I missing something?

If you need me to I can host an instance of GeoServer for you to test against.

Thank you!

**Comment by tomkralidis** [comment]
@mlacayoemery thanks for the report.  Sure if you can point me to a WPS instance to test that would be valuable.

**Comment by tomkralidis** [comment]
Thanks @mlacayoemery.  I've narrowed it down to:

```diff
diff --git a/owslib/wps.py b/owslib/wps.py
index b6c4736..7b6ed59 100644
--- a/owslib/wps.py
+++ b/owslib/wps.py
@@ -1591,7 +1591,7 @@ class GMLMultiPolygonFeatureCollection(FeatureCollection):
         dataElement = etree.Element(nspath_eval('wps:Data', namespaces))
         complexDataElement = etree.SubElement(
             dataElement, nspath_eval('wps:ComplexData', namespaces),
-                                              attrib={"mimeType": "text/xml", "encoding": "UTF-8", "schema": GML_SCHEMA_LOCATION})
+                                              attrib={"mimeType": "text/xml", "schema": GML_SCHEMA_LOCATION})
         featureMembersElement = etree.SubElement(
             complexDataElement, nspath_eval('gml:featureMembers', namespaces),
                                                  attrib={nspath_eval("xsi:schemaLocation", namespaces): "%s %s" % (DRAW_NAMESPACE, DRAW_SCHEMA_LOCATION)})
```

The encoding exception goes away however there is another (unrelated?) error. I don't have time to test further so if this works feel free to test further/issue a pull request.

**Comment by mlacayoemery** [comment]
Yes that does seem to fix that problem and I created a [pull request](https://github.com/geopython/OWSLib/pull/420). Have I done what you wanted correctly?

The new error seems to be an unrelated but similar specification implementation error, for which I created issue #419
